### PR TITLE
ChangeShip command and tests

### PR DESCRIPTION
**Pull request by Mpcs**
# Description
Adds a new console command "changeship".
All it does is change actual ship. It accepts one argument which is an ID of the target ship.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by Adrijaned** [comment]
@GooeyHub add to whitelist.

File: engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        console.warn("Usage: \"changeship module:shipName\"");
+    }
+
+    private SolShip cloneAndModifyShip(SolShip originalShip, String newShipID) {
+        HullConfig newHullConfig;
+        try {
+            newHullConfig = game.getHullConfigManager().getConfig(newShipID);
+        }
+        catch (RuntimeException e) {
+            return null;
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]: Could you use `Optional<SolShip>` instead?

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

File: engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java (hunk 1 above)
Comment (Mpcs) [reply engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]: Yes, changed it in the new commit

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public void handle(String input, Console console) {
+        String[] args = input.split(" ", 2);
+
+        if(args.length != 2) {
+            printHelp(console);
+            return;
+        }
+
+        Optional<SolShip> newShip = cloneAndModifyShip(hero.getShip(), args[1]);
+        if(!newShip.isPresent()) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]:
```suggestion
        if (!newShip.isPresent()) {
```

File: engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    private Optional<SolShip> cloneAndModifyShip(SolShip originalShip, String newShipID) {
+        HullConfig newHullConfig;
+        try {
+            HullConfigManager hcm = game.getHullConfigManager();
+            newHullConfig = hcm.getConfig(newShipID);
+        }
+        catch (RuntimeException e) {
+            System.out.println(e.getMessage());
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]: Please use the logger instead of System.out

Comment (Adrijaned) [reply engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]: You could also go around this whole try/catch by using `Assets.getDefaultAssetHelper().listJSONByRegex()` to ensure the existence of the ship, and `Validater.getValidatedJSON()` wit hthe proper schema should ensure you get really a ship. If it excepts even afterward, it's almost definitely a broken ship then, and you should crash in that case.

Comment (Adrijaned) [reply engine/src/main/java/org/destinationsol/game/console/commands/ChangeShipCommandHandler.java]: Please note - I have no idea how are the functions and classes mentioned in the above comment actually named.  The names I wrote just look righty-ish. You should be able to find them though if you can look around a bit.

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**